x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/pipeline_processors/create_ml_inference_pipeline.ts

```diff
@@ -91,15 +97,22 @@ export const createMlInferencePipeline = async (
     throw new Error(ErrorCode.PIPELINE_ALREADY_EXISTS);
   }
 
-  // Generate pipeline with default processors
-  const mlInferencePipeline = await formatMlPipelineBody(
-    inferencePipelineGeneratedName,
-    modelId,
-    sourceField,
-    destinationField || formatPipelineName(pipelineName),
-    inferenceConfig,
-    esClient
-  );
+  let mlInferencePipeline: MlInferencePipeline;
```

Comment:
nitpick: We try to avoid mutable variables as much as possible, and you can do this with early returns:
```
if (!(modelId && sourceField) && !pipelineDefinition) {
  throw new Error(ErrorCode.PARAMETER_CONFLICT);
}
const mlInferencePipeline = modelId && sourceField ? await formatMlPipelineBody(
      inferencePipelineGeneratedName,
      modelId,
      sourceField,
      destinationField || formatPipelineName(pipelineName),
      inferenceConfig,
      esClient
    ) : {...pipelineDefinition, version: 1 };
```

Reply:
Good to know. I avoided a ternary expression because it would have been so long, TIL that's preferable.